lib/Sema/CSApply.cpp

```diff
@@ -2514,6 +2514,65 @@ namespace {
             /*implicit=*/expr->isImplicit(), Type(), getType);
         result = finishApply(apply, Type(), cs.getConstraintLocator(expr));
       }
+			
+      // If the base type is optional and we're referring to an nominal
+      // type member via the dot syntax and the member name matches
+      // Optional<T>.none then emit a diagnostic
+      if (auto baseUnwrappedType = baseTy->lookThroughAllOptionalTypes()) {
```

Comment:
Since this method is already getting pretty long, can you split off the new functionality into a new method, such as diagnoseAmbiguousEnumMember() or similar?

Reply:
Done!